### DISCUSSION TITLE
Are photo and video sub-types of message?

### DISCUSSION THREAD

**Discussion by jsjoeio** [telegraf/telegraf#1484] [Q&A] [answered]
I tried looking through the docs or the codebase to figure this out but haven't had much luck.

My bot listens for messages in a group chat and then does things. I thought I could use one handler 
```javascript
bot.on("message", async (ctx) => {
  await handleOnMessage(ctx)
}
```

but it's not being triggered when messages have photos or videos attached. So my question is are `photo` and `video` sub-types of `message` and is there is a bug in my code? Or do I need to use separate listeners for that (i.e.

```javascript
bot.on("message", async (ctx) => {
  await handleOnMessage(ctx)
}
bot.on("photo", async (ctx) => {
  await handleOnMessage(ctx)
}
bot.on("video", async (ctx) => {
  await handleOnMessage(ctx)
}
```

**Comment by jsjoeio** [comment, marked as answer]
> is there is a bug in my code

I think this was the answer. I was looking for `message?.text` on `photo` and `video` updates. Turns out instead I needed to listen for `message?.caption` and now it works!

<img width="592" alt="Screen Shot 2021-07-04 at 8 36 00 AM" src="https://user-images.githubusercontent.com/3806031/124391009-36acb400-dca3-11eb-80b3-6cd36b401575.png">

**Comment by nzackoya** [comment]
I don't know what is wrong, but the message has no caption, or the firebase fucntions just skips the error message

**Reply by MKRhere** [reply]
Please share sample code that causes your issue. Also in the future avoid necroposting, and create a new discussion if the old one was already resolved and you still have an issue.
